src/rpc-common.ts

```diff
@@ -27,12 +29,14 @@ export abstract class RpcCommon implements IRpc {
   abstract sendResponse(id: number, response: any, success?: boolean): void;
   protected promiseCallbacks: Map<number, IPromiseCallbacks>; // promise resolve and reject callbacks that are called when returning from remote
   protected methods: Map<string, IMethod>;
+  protected logger: IChildLogger;
   // TODO: timeouts do not make sense for user interactions. consider not using timeouts by default
   protected timeout: number = 3600000; // timeout for response from remote in milliseconds
 
-  constructor() {
+  constructor(logger: IChildLogger) {
     this.promiseCallbacks = new Map();
     this.methods = new Map();
+    this.logger = logger;
```

Comment:
It is preferable to create a logger for each class with logger.createChildLogger({"className"}).

Reply:
done